vespalib/src/vespa/vespalib/stllike/cache.hpp

```diff
@@ -94,16 +93,8 @@ cache<P>::SizeConstrainedLru::try_get_and_ref(const KeyT& key, ValueT& val_out)
 
 template <typename P>
 void
-cache<P>::SizeConstrainedLru::evict_all() {
-    // There's no `clear()` on lrucache_map, so do it the hard way.
-    auto iter = Lru::begin();
-    while (iter != Lru::end()) {
-        // Entries will not be transitioned out of the cache via the probationary segment,
-        // so invoke the removal callback directly.
-        _owner.onRemove(iter.key());
-        iter = Lru::erase(iter); // This does _not_ invoke `removeOldest()`
-    }
-    set_size_bytes(0);
+cache<P>::SizeConstrainedLru::trim() {
+    Lru::trim();
```

Comment:
isn't this a subclass and would do this already?

Reply:
Yes—the parent class is `protected`, but I'll bring `Lru::trim` explicitly into scope instead.